binance/websocket/binance_socket_manager.py

```diff
@@ -1,83 +1,100 @@
-    ):
-        if stream_name in self._conns:
-            return False
+        self.on_message = on_message
+        self.on_open = on_open
+        self.on_close = on_close
+        self.on_ping = on_ping
+        self.on_pong = on_pong
+        self.on_error = on_error
+        self.create_ws_connection()
 
-        if is_combined:
-            factory_url = self.stream_url + "/stream"
-        else:
-            factory_url = self.stream_url + "/ws"
-
-        if not is_live:
-            payload_obj = json.loads(payload.decode("utf8"))
-
-            if is_combined:
-                factory_url = factory_url + "?streams=" + payload_obj["params"]
-            else:
-                factory_url = factory_url + "/" + payload_obj["params"]
-            payload = None
+    def create_ws_connection(self):
+        self.logger.debug("Start connect to WebSocket Server: %s", self.stream_url)
+        self.ws = create_connection(self.stream_url)
+        self.logger.debug("Created WebSocket connection to: %s", self.stream_url)
+        self._callback(self.on_open)
 
-        self._logger.info("Connection with URL: {}".format(factory_url))
-
-        factory = BinanceClientFactory(factory_url, payload=payload)
-        factory.base_client = self
-        factory.protocol = BinanceClientProtocol
-        factory.setProtocolOptions(
-            openHandshakeTimeout=5, autoPingInterval=300, autoPingTimeout=5
-        )
-        factory.callback = callback
-        self.factories[stream_name] = factory
-        reactor.callFromThread(self.add_connection, stream_name, self.stream_url)
-
-    def add_connection(self, stream_name, url):
-        if not url.startswith("wss://"):
-            raise ValueError("expected wss:// URL prefix")
+    def run(self):
+        self.read_data()
 
-        factory = self.factories[stream_name]
-        options = ssl.optionsForClientTLS(hostname=urlparse(url).hostname)
-        self._conns[stream_name] = connectWS(factory, options)
+    def send_message(self, message):
+        self.logger.debug("Sending message to Binance WebSocket Server: %s", message)
+        self.ws.send(message)
 
-    def stop_socket(self, conn_key):
-        if conn_key not in self._conns:
-            return
+    def ping(self):
+        self.ws.ping()
 
-        # disable reconnecting if we are closing
-        self._conns[conn_key].factory = WebSocketClientFactory(self.stream_url)
-        self._conns[conn_key].disconnect()
-        del self._conns[conn_key]
+    def read_data(self):
+        data = ""
+        while True:
+            try:
+                op_code, frame = self.ws.recv_data_frame(True)
+            except WebSocketException as e:
+                if isinstance(e, WebSocketConnectionClosedException):
+                    self.logger.error("lost websocket connection")
+                else:
+                    self.logger.error("websocket exception: %s", e)
```

Comment:
A bit nitpick, but sometimes you use % formatting for strings, and in others .format . I would keep the same everywhere just for consistency with the style.

In the logs, sometimes you start with uppercase and others lowercase.